crates/generated_parser/src/ast_builder.rs

```diff
@@ -4799,6 +4969,34 @@ impl<'alloc> AstBuilder<'alloc> {
         Ok(())
     }
 
+    fn check_unhandled_break_or_continue<T>(
+        &mut self,
+        context: T,
+        offset: usize,
+    ) -> Result<'alloc, ()>
+    where
+        T: ControlEarlyErrorsContext,
+    {
+        let index = self.find_first_break_or_continue(offset);
+        if let Some(info) = self.breaks_and_continues.get(index.index) {
```

Comment:
this needs to be wrapped in a method, placed near `find_first_break_or_continue`
if the caller unwraps Index type, it doesn't improve the safety,
so, if the caller receives Index type, it should feed it to another method without unwrapping,
so that the type is checked at the method parameter type.